File: blackjack.py
from poker import Deck, Card, Player
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print("                  Welcome to BlackJack!")
print()
print("                           $")
print("                         $$$$$")
print("                       $$  $  $$")
print("                      $    $    $")
print("                      $    $    ")
print("                       $$  $  ")
print("                         $$$$$")
print("                           $  $$")
print("                           $    $")
print("                      $    $    $")
print("                       $$  $  $$")
print("                         $$$$$")
print("                           $")
print()

playing = True

while playing:
    myDeck = Deck()
    myDeck.shuffle()
    dealer = Player()
    player = Player()
    myDeck.deal(dealer, 2)
    myDeck.deal(player, 2)
    print("Here are the dealer's cards: " + dealer.__str__())
    dealer_sum = dealer.get_value()
    print("They add to " + str(dealer_sum))
    print()
    print("Here are your cards: " + player.__str__())
    player_sum = player.get_value()
    print("They add to " + str(player_sum))
    print()
    hitting = True
    while hitting:
        answer = input("Hit or Stay?")
        if answer.lower() == "hit":
            myDeck.deal(player, 1)
            print("Here are your cards: " + player.__str__())
            player_sum = player.get_value()
            if player_sum > 21:
                print("BUST!")
                print()
                hitting = False
            else:
                print("They add to " + str(player_sum))
                print()
        else:
            hitting = False
    while dealer_sum < 17 and player_sum <= 21:
        print()
        print("The dealer hits.")
        myDeck.deal(dealer,1)
        print("Here are the dealer's cards: " + dealer.__str__())
        dealer_sum = dealer.get_value()
        print("They add to " + str(dealer_sum))
        print()
    if dealer_sum > 21:
        print("The dealer busts!")
        print()
    elif player_sum <= 21:
        print("The dealer stays.")
        print()
    if 21 >= player_sum > dealer_sum or dealer_sum > 21:
        print("You won!")
        print()
    elif player_sum == dealer_sum:
        print("You tied with the dealer.")
        print()
    else:
        print("You lost. :(")
        print()
    if str(input("Would you like to play again (Yes or No)?")).lower() != "yes":
        playing = False
    print()
    print()

File: java/Blackjack.java
import java.util.*;

public class Blackjack {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in).useDelimiter("\\s");
		boolean playing = true;
		while (playing) {
			Player dealer = new Player();
			Player player = new Player();
			Deck deck = new Deck();
			deck.shuffle();
			deck.deal(dealer, 2);
			deck.deal(player, 2);
			System.out.println("Here are the dealers cards: " + dealer.toString());
			System.out.println("They add to " + dealer.getValue());
			System.out.println();
			System.out.println("Here are your cards: " + player.toString());
			System.out.println("They add to " + player.getValue());
			System.out.println();
			boolean hitting = true;
			while (hitting) {
				System.out.println("Hit or stay?");
				String answer = input.next();
				System.out.println();
				answer = answer.toLowerCase();
				if (answer.equals("hit")) {
					deck.deal(player, 1);
					System.out.println("Here are your cards: " + player.toString());
					System.out.println("They add to " + player.getValue());
					System.out.println();
					if (player.getValue() > 21) {
						hitting = false;
						System.out.println("Bust! You lost. :(");
					}
				} else {
					hitting = false;
				}
			}
			while (dealer.getValue() <= 17 && player.getValue() <= 21) {
				deck.deal(dealer, 1);
				System.out.println("Here are the dealer's cards: " + dealer.toString());
				System.out.println("They add to: " + dealer.getValue());
				System.out.println();
			}
			if (dealer.getValue() > 21) {
				System.out.println("The dealer busts! You win! :)");
				System.out.println();
			} else if (dealer.getValue() < player.getValue() && player.getValue() <= 21) {
				System.out.println("You win! :)");
				System.out.println();
			} else if (dealer.getValue() == player.getValue()) {
				System.out.println("You tied with the dealer. :|");
				System.out.println();
			} else {
				System.out.println("You lost. :(");
				System.out.println();
			}
			System.out.println("Would you like to play again? (yes or no)");
			String answer = input.next();
			answer = answer.toLowerCase();
			if (!answer.equals("yes")) {
				playing = false;
			}
			System.out.println();
			System.out.println();
			System.out.println();
		}
	}
}
